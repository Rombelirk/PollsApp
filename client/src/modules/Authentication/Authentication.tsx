import React, { FC, useState, FormEvent, MouseEventHandler } from 'react';
import { AuthForm } from './Authentication.styles';
import Input from '../../atoms/Input/Input';
import Button from '../../atoms/Button/Button';

enum AuthCredentials {
    login = 'login',
    password = 'password',
}

interface Props {
    onFormSubmit: (creds: { login: string; password: string }) => void;
}

const Authentication: FC<Props> = ({ onFormSubmit }) => {
    const [login, setLogin] = useState<string>('');
    const [password, setPassword] = useState<string>('');

    const onFieldChange = (type: AuthCredentials) => (event: FormEvent<HTMLInputElement>) => {
        if (type === AuthCredentials.login) {
            return setLogin(event.currentTarget.value);
        }
        return setPassword(event.currentTarget.value);
    };

    const onSubmit: MouseEventHandler<HTMLElement> = (event) => {
        onFormSubmit({ login, password });
    };

    return (
        <AuthForm>
            <Input value={login} onChange={onFieldChange(AuthCredentials.login)} />
            <Input value={password} onChange={onFieldChange(AuthCredentials.password)} />
            <Button onClick={onSubmit}>Submit</Button>
        </AuthForm>
    );
};

export default Authentication;
