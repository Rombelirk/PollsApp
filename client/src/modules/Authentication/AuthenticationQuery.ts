
import gql from 'graphql-tag';

const authQuery = gql`
    query AuthenticationQuery($input: LoginInput!) {
        login(input: $input) {
            jwtToken
            user {
                _id
                login
            }
        }
    }
`;

export default authQuery;
