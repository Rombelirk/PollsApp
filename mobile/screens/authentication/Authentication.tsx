import React, { useState, FC } from 'react';
import { Text } from 'react-native';
import { Container, InputContainer, Form, TextInput, SubmitButton } from './Authentication.styles';

interface Props {
    onFormSubmit: (credentials: { login: string; password: string }) => void;
}

enum Credentials {
    login = 'login',
    password = 'password'
}

const Authentication: FC<Props> = ({ onFormSubmit }) => {

    const [login, setLogin] = useState("");
    const [password, setPassword] = useState("");

    const onSubmit = e => {
        onFormSubmit({ login, password });
    }

    const onInputChange = (type: string) => (text: string) => {
        if (type === Credentials.login) {
            setLogin(text)
        } else if (Credentials.password) {
            setPassword(text)
        }
    }

    return (
        <Container>
            <Form>
                <InputContainer>
                    <Text>Login</Text>
                    <TextInput onChangeText={onInputChange(Credentials.login)} value={login} />
                </InputContainer>
                <InputContainer>
                    <Text>Password</Text>
                    <TextInput onChangeText={onInputChange(Credentials.password)} password={true} />
                </InputContainer>
                <InputContainer>
                    <SubmitButton title="Submit" onPress={onSubmit} />
                </InputContainer>
            </Form>
        </Container>
    )
}

export default Authentication;

