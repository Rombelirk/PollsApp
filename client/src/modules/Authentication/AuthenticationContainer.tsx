import React, { FC } from 'react';
import authQuery from './AuthenticationQuery';
import Authentication from './Authentication';
import client from '../../client';
import { LoginInput } from '../../../__generated__/globalTypes';
import { AuthenticationQuery } from './__generated__/AuthenticationQuery';
import { withRouter, RouteComponentProps } from 'react-router';

interface Props extends RouteComponentProps {}

const AuthenticationContainer: FC<Props> = ({ history }) => {
    const onFormSubmit = async (creds: { login: string; password: string }) => {
        const { login, password } = creds;
        try {
            const result = await client.query<AuthenticationQuery, LoginInput>({
                query: authQuery,
                variables: {
                    login,
                    password,
                },
            });

            if (result.data.login.jwtToken) {
                localStorage.setItem('jwtToken', result.data.login.jwtToken);
                history.push('/');
            }
        } catch (e) {
            console.log(e);
        }
    };

    return <Authentication onFormSubmit={onFormSubmit} />;
};

export default withRouter(AuthenticationContainer);
