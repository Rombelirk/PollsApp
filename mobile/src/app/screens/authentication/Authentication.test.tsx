import React from 'react';
import Authentication from './Authentication'
import renderer from 'react-test-renderer';

describe('<Authentication />', () => {
    test('renders correctly', () => {
        const tree = renderer.create(<Authentication />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});