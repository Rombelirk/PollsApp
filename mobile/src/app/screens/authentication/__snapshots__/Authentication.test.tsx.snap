// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 80,
          "width": 80,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <Text>
        Login
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRadius": 3,
              "borderWidth": 1,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <Text>
        Password
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        password={true}
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRadius": 3,
              "borderWidth": 1,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
